// @devvit/protos/json/reddit/devvit/post/v1/post.js
/**
 * #post.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
/** The style of the post's rendering. */
export var RenderStyle;
(function (RenderStyle) {
    /** RENDER_STYLE_UNSPECIFIED - The style of the post's rendering is not specified. */
    RenderStyle[RenderStyle["RENDER_STYLE_UNSPECIFIED"] = 0] = "RENDER_STYLE_UNSPECIFIED";
    /** RENDER_STYLE_DEFAULT - The style of the post's rendering is the default style. */
    RenderStyle[RenderStyle["RENDER_STYLE_DEFAULT"] = 1] = "RENDER_STYLE_DEFAULT";
    /**
     * RENDER_STYLE_CHROMELESS - The style of the post's rendering is the chromeless style,
     * which should be rendered without the post header, footer, and side margins.
     */
    RenderStyle[RenderStyle["RENDER_STYLE_CHROMELESS"] = 2] = "RENDER_STYLE_CHROMELESS";
    RenderStyle[RenderStyle["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(RenderStyle || (RenderStyle = {}));
/** The height of the Post entrypoint. This is not expected to change often. */
export var EntrypointHeight;
(function (EntrypointHeight) {
    EntrypointHeight[EntrypointHeight["HEIGHT_UNSPECIFIED"] = 0] = "HEIGHT_UNSPECIFIED";
    EntrypointHeight[EntrypointHeight["REGULAR"] = 1] = "REGULAR";
    EntrypointHeight[EntrypointHeight["TALL"] = 2] = "TALL";
    EntrypointHeight[EntrypointHeight["SHORT"] = 3] = "SHORT";
    EntrypointHeight[EntrypointHeight["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(EntrypointHeight || (EntrypointHeight = {}));
/**
 * Matches the values in the MediaService.MimeType enum.
 * See: https://github.snooguts.net/reddit/media-api/blob/master/api/protos/reddit/foundation/media/v1/media.proto
 */
export var ImageDataMimeType;
(function (ImageDataMimeType) {
    ImageDataMimeType[ImageDataMimeType["GIF"] = 0] = "GIF";
    ImageDataMimeType[ImageDataMimeType["JPEG"] = 1] = "JPEG";
    ImageDataMimeType[ImageDataMimeType["PNG"] = 2] = "PNG";
    ImageDataMimeType[ImageDataMimeType["WEBP"] = 3] = "WEBP";
    ImageDataMimeType[ImageDataMimeType["MP4"] = 4] = "MP4";
    ImageDataMimeType[ImageDataMimeType["QUICKTIME"] = 5] = "QUICKTIME";
    ImageDataMimeType[ImageDataMimeType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(ImageDataMimeType || (ImageDataMimeType = {}));
