// @devvit/protos/types/devvit/dev_portal/app_account_migration/app_account_migration.js
/**
 * #app_account_migration.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
import _m0 from "protobufjs/minimal.js";
import { messageTypeRegistry } from "../../../typeRegistry.js";
function createBaseConvertBotAccountToAppAccountRequest() {
    return { appId: "", botAccountId: "", botAccountPassword: "" };
}
export const ConvertBotAccountToAppAccountRequest = {
    $type: "devvit.dev_portal.app_account_migration.ConvertBotAccountToAppAccountRequest",
    encode(message, writer = _m0.Writer.create()) {
        if (message.appId !== "") {
            writer.uint32(10).string(message.appId);
        }
        if (message.botAccountId !== "") {
            writer.uint32(18).string(message.botAccountId);
        }
        if (message.botAccountPassword !== "") {
            writer.uint32(26).string(message.botAccountPassword);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseConvertBotAccountToAppAccountRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.appId = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.botAccountId = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.botAccountPassword = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            appId: isSet(object.appId) ? globalThis.String(object.appId) : "",
            botAccountId: isSet(object.botAccountId) ? globalThis.String(object.botAccountId) : "",
            botAccountPassword: isSet(object.botAccountPassword) ? globalThis.String(object.botAccountPassword) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.appId !== "") {
            obj.appId = message.appId;
        }
        if (message.botAccountId !== "") {
            obj.botAccountId = message.botAccountId;
        }
        if (message.botAccountPassword !== "") {
            obj.botAccountPassword = message.botAccountPassword;
        }
        return obj;
    },
    create(base) {
        return ConvertBotAccountToAppAccountRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseConvertBotAccountToAppAccountRequest();
        message.appId = object.appId ?? "";
        message.botAccountId = object.botAccountId ?? "";
        message.botAccountPassword = object.botAccountPassword ?? "";
        return message;
    },
};
messageTypeRegistry.set(ConvertBotAccountToAppAccountRequest.$type, ConvertBotAccountToAppAccountRequest);
function createBaseConvertBotAccountToAppAccountResponse() {
    return { appId: "", previousAppSlug: "", newAppSlug: "" };
}
export const ConvertBotAccountToAppAccountResponse = {
    $type: "devvit.dev_portal.app_account_migration.ConvertBotAccountToAppAccountResponse",
    encode(message, writer = _m0.Writer.create()) {
        if (message.appId !== "") {
            writer.uint32(10).string(message.appId);
        }
        if (message.previousAppSlug !== "") {
            writer.uint32(18).string(message.previousAppSlug);
        }
        if (message.newAppSlug !== "") {
            writer.uint32(26).string(message.newAppSlug);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseConvertBotAccountToAppAccountResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.appId = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.previousAppSlug = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.newAppSlug = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            appId: isSet(object.appId) ? globalThis.String(object.appId) : "",
            previousAppSlug: isSet(object.previousAppSlug) ? globalThis.String(object.previousAppSlug) : "",
            newAppSlug: isSet(object.newAppSlug) ? globalThis.String(object.newAppSlug) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.appId !== "") {
            obj.appId = message.appId;
        }
        if (message.previousAppSlug !== "") {
            obj.previousAppSlug = message.previousAppSlug;
        }
        if (message.newAppSlug !== "") {
            obj.newAppSlug = message.newAppSlug;
        }
        return obj;
    },
    create(base) {
        return ConvertBotAccountToAppAccountResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseConvertBotAccountToAppAccountResponse();
        message.appId = object.appId ?? "";
        message.previousAppSlug = object.previousAppSlug ?? "";
        message.newAppSlug = object.newAppSlug ?? "";
        return message;
    },
};
messageTypeRegistry.set(ConvertBotAccountToAppAccountResponse.$type, ConvertBotAccountToAppAccountResponse);
function isSet(value) {
    return value !== null && value !== undefined;
}
