// @devvit/protos/types/devvit/plugin/externalendpoints/v1alpha/externalendpoints_internal.js
/**
 * #externalendpoints_internal.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
import _m0 from "protobufjs/minimal.js";
import { Timestamp } from "../../../../google/protobuf/timestamp.js";
import { messageTypeRegistry } from "../../../../typeRegistry.js";
function createBaseCallbackContext() {
    return {
        userT2: "",
        installationId: "",
        subredditT5: "",
        postT3: "",
        commentT1: "",
        userLoid: "",
        acceptLanguage: "",
        acceptTimezone: "",
    };
}
export const CallbackContext = {
    $type: "devvit.plugin.externalendpoints.v1alpha.CallbackContext",
    encode(message, writer = _m0.Writer.create()) {
        if (message.userT2 !== "") {
            writer.uint32(10).string(message.userT2);
        }
        if (message.installationId !== "") {
            writer.uint32(18).string(message.installationId);
        }
        if (message.subredditT5 !== "") {
            writer.uint32(26).string(message.subredditT5);
        }
        if (message.postT3 !== "") {
            writer.uint32(34).string(message.postT3);
        }
        if (message.commentT1 !== "") {
            writer.uint32(42).string(message.commentT1);
        }
        if (message.userLoid !== "") {
            writer.uint32(58).string(message.userLoid);
        }
        if (message.acceptLanguage !== "") {
            writer.uint32(66).string(message.acceptLanguage);
        }
        if (message.acceptTimezone !== "") {
            writer.uint32(74).string(message.acceptTimezone);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseCallbackContext();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.userT2 = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.installationId = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.subredditT5 = reader.string();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.postT3 = reader.string();
                    continue;
                case 5:
                    if (tag !== 42) {
                        break;
                    }
                    message.commentT1 = reader.string();
                    continue;
                case 7:
                    if (tag !== 58) {
                        break;
                    }
                    message.userLoid = reader.string();
                    continue;
                case 8:
                    if (tag !== 66) {
                        break;
                    }
                    message.acceptLanguage = reader.string();
                    continue;
                case 9:
                    if (tag !== 74) {
                        break;
                    }
                    message.acceptTimezone = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            userT2: isSet(object.userT2) ? globalThis.String(object.userT2) : "",
            installationId: isSet(object.installationId) ? globalThis.String(object.installationId) : "",
            subredditT5: isSet(object.subredditT5) ? globalThis.String(object.subredditT5) : "",
            postT3: isSet(object.postT3) ? globalThis.String(object.postT3) : "",
            commentT1: isSet(object.commentT1) ? globalThis.String(object.commentT1) : "",
            userLoid: isSet(object.userLoid) ? globalThis.String(object.userLoid) : "",
            acceptLanguage: isSet(object.acceptLanguage) ? globalThis.String(object.acceptLanguage) : "",
            acceptTimezone: isSet(object.acceptTimezone) ? globalThis.String(object.acceptTimezone) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.userT2 !== "") {
            obj.userT2 = message.userT2;
        }
        if (message.installationId !== "") {
            obj.installationId = message.installationId;
        }
        if (message.subredditT5 !== "") {
            obj.subredditT5 = message.subredditT5;
        }
        if (message.postT3 !== "") {
            obj.postT3 = message.postT3;
        }
        if (message.commentT1 !== "") {
            obj.commentT1 = message.commentT1;
        }
        if (message.userLoid !== "") {
            obj.userLoid = message.userLoid;
        }
        if (message.acceptLanguage !== "") {
            obj.acceptLanguage = message.acceptLanguage;
        }
        if (message.acceptTimezone !== "") {
            obj.acceptTimezone = message.acceptTimezone;
        }
        return obj;
    },
    create(base) {
        return CallbackContext.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseCallbackContext();
        message.userT2 = object.userT2 ?? "";
        message.installationId = object.installationId ?? "";
        message.subredditT5 = object.subredditT5 ?? "";
        message.postT3 = object.postT3 ?? "";
        message.commentT1 = object.commentT1 ?? "";
        message.userLoid = object.userLoid ?? "";
        message.acceptLanguage = object.acceptLanguage ?? "";
        message.acceptTimezone = object.acceptTimezone ?? "";
        return message;
    },
};
messageTypeRegistry.set(CallbackContext.$type, CallbackContext);
function createBaseConsumeCallbackTokenRequest() {
    return { tokenId: "" };
}
export const ConsumeCallbackTokenRequest = {
    $type: "devvit.plugin.externalendpoints.v1alpha.ConsumeCallbackTokenRequest",
    encode(message, writer = _m0.Writer.create()) {
        if (message.tokenId !== "") {
            writer.uint32(10).string(message.tokenId);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseConsumeCallbackTokenRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.tokenId = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { tokenId: isSet(object.tokenId) ? globalThis.String(object.tokenId) : "" };
    },
    toJSON(message) {
        const obj = {};
        if (message.tokenId !== "") {
            obj.tokenId = message.tokenId;
        }
        return obj;
    },
    create(base) {
        return ConsumeCallbackTokenRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseConsumeCallbackTokenRequest();
        message.tokenId = object.tokenId ?? "";
        return message;
    },
};
messageTypeRegistry.set(ConsumeCallbackTokenRequest.$type, ConsumeCallbackTokenRequest);
function createBaseConsumeCallbackTokenResponse() {
    return {};
}
export const ConsumeCallbackTokenResponse = {
    $type: "devvit.plugin.externalendpoints.v1alpha.ConsumeCallbackTokenResponse",
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseConsumeCallbackTokenResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return ConsumeCallbackTokenResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseConsumeCallbackTokenResponse();
        return message;
    },
};
messageTypeRegistry.set(ConsumeCallbackTokenResponse.$type, ConsumeCallbackTokenResponse);
function createBaseCreateManagedTokenRequest() {
    return { tokenName: "", appSlug: "" };
}
export const CreateManagedTokenRequest = {
    $type: "devvit.plugin.externalendpoints.v1alpha.CreateManagedTokenRequest",
    encode(message, writer = _m0.Writer.create()) {
        if (message.tokenName !== "") {
            writer.uint32(10).string(message.tokenName);
        }
        if (message.appSlug !== "") {
            writer.uint32(18).string(message.appSlug);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseCreateManagedTokenRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.tokenName = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.appSlug = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            tokenName: isSet(object.tokenName) ? globalThis.String(object.tokenName) : "",
            appSlug: isSet(object.appSlug) ? globalThis.String(object.appSlug) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.tokenName !== "") {
            obj.tokenName = message.tokenName;
        }
        if (message.appSlug !== "") {
            obj.appSlug = message.appSlug;
        }
        return obj;
    },
    create(base) {
        return CreateManagedTokenRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseCreateManagedTokenRequest();
        message.tokenName = object.tokenName ?? "";
        message.appSlug = object.appSlug ?? "";
        return message;
    },
};
messageTypeRegistry.set(CreateManagedTokenRequest.$type, CreateManagedTokenRequest);
function createBaseCreateManagedTokenResponse() {
    return { token: undefined };
}
export const CreateManagedTokenResponse = {
    $type: "devvit.plugin.externalendpoints.v1alpha.CreateManagedTokenResponse",
    encode(message, writer = _m0.Writer.create()) {
        if (message.token !== undefined) {
            Token.encode(message.token, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseCreateManagedTokenResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.token = Token.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { token: isSet(object.token) ? Token.fromJSON(object.token) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.token !== undefined) {
            obj.token = Token.toJSON(message.token);
        }
        return obj;
    },
    create(base) {
        return CreateManagedTokenResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseCreateManagedTokenResponse();
        message.token = (object.token !== undefined && object.token !== null) ? Token.fromPartial(object.token) : undefined;
        return message;
    },
};
messageTypeRegistry.set(CreateManagedTokenResponse.$type, CreateManagedTokenResponse);
function createBaseListTokensRequest() {
    return { appSlug: "" };
}
export const ListTokensRequest = {
    $type: "devvit.plugin.externalendpoints.v1alpha.ListTokensRequest",
    encode(message, writer = _m0.Writer.create()) {
        if (message.appSlug !== "") {
            writer.uint32(10).string(message.appSlug);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseListTokensRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.appSlug = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { appSlug: isSet(object.appSlug) ? globalThis.String(object.appSlug) : "" };
    },
    toJSON(message) {
        const obj = {};
        if (message.appSlug !== "") {
            obj.appSlug = message.appSlug;
        }
        return obj;
    },
    create(base) {
        return ListTokensRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseListTokensRequest();
        message.appSlug = object.appSlug ?? "";
        return message;
    },
};
messageTypeRegistry.set(ListTokensRequest.$type, ListTokensRequest);
function createBaseListTokensResponse() {
    return { tokens: [] };
}
export const ListTokensResponse = {
    $type: "devvit.plugin.externalendpoints.v1alpha.ListTokensResponse",
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.tokens) {
            Token.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseListTokensResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.tokens.push(Token.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { tokens: globalThis.Array.isArray(object?.tokens) ? object.tokens.map((e) => Token.fromJSON(e)) : [] };
    },
    toJSON(message) {
        const obj = {};
        if (message.tokens?.length) {
            obj.tokens = message.tokens.map((e) => Token.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return ListTokensResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseListTokensResponse();
        message.tokens = object.tokens?.map((e) => Token.fromPartial(e)) || [];
        return message;
    },
};
messageTypeRegistry.set(ListTokensResponse.$type, ListTokensResponse);
function createBaseRevokeTokenRequest() {
    return { tokenId: "" };
}
export const RevokeTokenRequest = {
    $type: "devvit.plugin.externalendpoints.v1alpha.RevokeTokenRequest",
    encode(message, writer = _m0.Writer.create()) {
        if (message.tokenId !== "") {
            writer.uint32(10).string(message.tokenId);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseRevokeTokenRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.tokenId = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { tokenId: isSet(object.tokenId) ? globalThis.String(object.tokenId) : "" };
    },
    toJSON(message) {
        const obj = {};
        if (message.tokenId !== "") {
            obj.tokenId = message.tokenId;
        }
        return obj;
    },
    create(base) {
        return RevokeTokenRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseRevokeTokenRequest();
        message.tokenId = object.tokenId ?? "";
        return message;
    },
};
messageTypeRegistry.set(RevokeTokenRequest.$type, RevokeTokenRequest);
function createBaseRevokeTokenResponse() {
    return {};
}
export const RevokeTokenResponse = {
    $type: "devvit.plugin.externalendpoints.v1alpha.RevokeTokenResponse",
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseRevokeTokenResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return RevokeTokenResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseRevokeTokenResponse();
        return message;
    },
};
messageTypeRegistry.set(RevokeTokenResponse.$type, RevokeTokenResponse);
function createBaseToken() {
    return { id: "", name: "", token: undefined, createdAt: undefined, createdByT2: "" };
}
export const Token = {
    $type: "devvit.plugin.externalendpoints.v1alpha.Token",
    encode(message, writer = _m0.Writer.create()) {
        if (message.id !== "") {
            writer.uint32(10).string(message.id);
        }
        if (message.name !== "") {
            writer.uint32(18).string(message.name);
        }
        if (message.token !== undefined) {
            writer.uint32(26).string(message.token);
        }
        if (message.createdAt !== undefined) {
            Timestamp.encode(toTimestamp(message.createdAt), writer.uint32(34).fork()).ldelim();
        }
        if (message.createdByT2 !== "") {
            writer.uint32(42).string(message.createdByT2);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseToken();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.id = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.name = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.token = reader.string();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.createdAt = fromTimestamp(Timestamp.decode(reader, reader.uint32()));
                    continue;
                case 5:
                    if (tag !== 42) {
                        break;
                    }
                    message.createdByT2 = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            id: isSet(object.id) ? globalThis.String(object.id) : "",
            name: isSet(object.name) ? globalThis.String(object.name) : "",
            token: isSet(object.token) ? globalThis.String(object.token) : undefined,
            createdAt: isSet(object.createdAt) ? fromJsonTimestamp(object.createdAt) : undefined,
            createdByT2: isSet(object.createdByT2) ? globalThis.String(object.createdByT2) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.id !== "") {
            obj.id = message.id;
        }
        if (message.name !== "") {
            obj.name = message.name;
        }
        if (message.token !== undefined) {
            obj.token = message.token;
        }
        if (message.createdAt !== undefined) {
            obj.createdAt = message.createdAt.toISOString();
        }
        if (message.createdByT2 !== "") {
            obj.createdByT2 = message.createdByT2;
        }
        return obj;
    },
    create(base) {
        return Token.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseToken();
        message.id = object.id ?? "";
        message.name = object.name ?? "";
        message.token = object.token ?? undefined;
        message.createdAt = object.createdAt ?? undefined;
        message.createdByT2 = object.createdByT2 ?? "";
        return message;
    },
};
messageTypeRegistry.set(Token.$type, Token);
function createBaseValidateTokenRequest() {
    return { token: "" };
}
export const ValidateTokenRequest = {
    $type: "devvit.plugin.externalendpoints.v1alpha.ValidateTokenRequest",
    encode(message, writer = _m0.Writer.create()) {
        if (message.token !== "") {
            writer.uint32(10).string(message.token);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseValidateTokenRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.token = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { token: isSet(object.token) ? globalThis.String(object.token) : "" };
    },
    toJSON(message) {
        const obj = {};
        if (message.token !== "") {
            obj.token = message.token;
        }
        return obj;
    },
    create(base) {
        return ValidateTokenRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseValidateTokenRequest();
        message.token = object.token ?? "";
        return message;
    },
};
messageTypeRegistry.set(ValidateTokenRequest.$type, ValidateTokenRequest);
function createBaseValidateTokenResponse() {
    return { appSlug: "", callbackContext: undefined, tokenId: "" };
}
export const ValidateTokenResponse = {
    $type: "devvit.plugin.externalendpoints.v1alpha.ValidateTokenResponse",
    encode(message, writer = _m0.Writer.create()) {
        if (message.appSlug !== "") {
            writer.uint32(10).string(message.appSlug);
        }
        if (message.callbackContext !== undefined) {
            CallbackContext.encode(message.callbackContext, writer.uint32(18).fork()).ldelim();
        }
        if (message.tokenId !== "") {
            writer.uint32(26).string(message.tokenId);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseValidateTokenResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.appSlug = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.callbackContext = CallbackContext.decode(reader, reader.uint32());
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.tokenId = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            appSlug: isSet(object.appSlug) ? globalThis.String(object.appSlug) : "",
            callbackContext: isSet(object.callbackContext) ? CallbackContext.fromJSON(object.callbackContext) : undefined,
            tokenId: isSet(object.tokenId) ? globalThis.String(object.tokenId) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.appSlug !== "") {
            obj.appSlug = message.appSlug;
        }
        if (message.callbackContext !== undefined) {
            obj.callbackContext = CallbackContext.toJSON(message.callbackContext);
        }
        if (message.tokenId !== "") {
            obj.tokenId = message.tokenId;
        }
        return obj;
    },
    create(base) {
        return ValidateTokenResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseValidateTokenResponse();
        message.appSlug = object.appSlug ?? "";
        message.callbackContext = (object.callbackContext !== undefined && object.callbackContext !== null)
            ? CallbackContext.fromPartial(object.callbackContext)
            : undefined;
        message.tokenId = object.tokenId ?? "";
        return message;
    },
};
messageTypeRegistry.set(ValidateTokenResponse.$type, ValidateTokenResponse);
export const ExternalEndpointsInternalServiceName = "devvit.plugin.externalendpoints.v1alpha.ExternalEndpointsInternal";
export class ExternalEndpointsInternalClientImpl {
    constructor(rpc, opts) {
        this.service = opts?.service || ExternalEndpointsInternalServiceName;
        this.rpc = rpc;
        this.CreateManagedToken = this.CreateManagedToken.bind(this);
        this.ValidateToken = this.ValidateToken.bind(this);
        this.ConsumeCallbackToken = this.ConsumeCallbackToken.bind(this);
        this.ListTokens = this.ListTokens.bind(this);
        this.RevokeToken = this.RevokeToken.bind(this);
    }
    CreateManagedToken(request, metadata) {
        const data = CreateManagedTokenRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "CreateManagedToken", data, metadata);
        return promise.then((data) => CreateManagedTokenResponse.decode(_m0.Reader.create(data)));
    }
    ValidateToken(request, metadata) {
        const data = ValidateTokenRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "ValidateToken", data, metadata);
        return promise.then((data) => ValidateTokenResponse.decode(_m0.Reader.create(data)));
    }
    ConsumeCallbackToken(request, metadata) {
        const data = ConsumeCallbackTokenRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "ConsumeCallbackToken", data, metadata);
        return promise.then((data) => ConsumeCallbackTokenResponse.decode(_m0.Reader.create(data)));
    }
    ListTokens(request, metadata) {
        const data = ListTokensRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "ListTokens", data, metadata);
        return promise.then((data) => ListTokensResponse.decode(_m0.Reader.create(data)));
    }
    RevokeToken(request, metadata) {
        const data = RevokeTokenRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "RevokeToken", data, metadata);
        return promise.then((data) => RevokeTokenResponse.decode(_m0.Reader.create(data)));
    }
}
export const ExternalEndpointsInternalDefinition = {
    name: "ExternalEndpointsInternal",
    fullName: "devvit.plugin.externalendpoints.v1alpha.ExternalEndpointsInternal",
    methods: {
        createManagedToken: {
            name: "CreateManagedToken",
            requestType: CreateManagedTokenRequest,
            requestStream: false,
            responseType: CreateManagedTokenResponse,
            responseStream: false,
            options: {},
        },
        /**
         * Validate a callback or managed token. Returns an empty response
         * (`token_id == ""`) if the token is invalid. Updates last used timestamp.
         */
        validateToken: {
            name: "ValidateToken",
            requestType: ValidateTokenRequest,
            requestStream: false,
            responseType: ValidateTokenResponse,
            responseStream: false,
            options: {},
        },
        /**
         * Deletes the token. Called by gateway after a successful app invocation. If
         * the app errored, gateway does not consume; the row stays alive and the
         * callback URL can be retried until its TTL expires. Intentional app failures
         * and concurrent requests are not prevented.
         */
        consumeCallbackToken: {
            name: "ConsumeCallbackToken",
            requestType: ConsumeCallbackTokenRequest,
            requestStream: false,
            responseType: ConsumeCallbackTokenResponse,
            responseStream: false,
            options: {},
        },
        /** List all tokens for an app. */
        listTokens: {
            name: "ListTokens",
            requestType: ListTokensRequest,
            requestStream: false,
            responseType: ListTokensResponse,
            responseStream: false,
            options: {},
        },
        revokeToken: {
            name: "RevokeToken",
            requestType: RevokeTokenRequest,
            requestStream: false,
            responseType: RevokeTokenResponse,
            responseStream: false,
            options: {},
        },
    },
};
function toTimestamp(date) {
    const seconds = Math.trunc(date.getTime() / 1000);
    const nanos = (date.getTime() % 1000) * 1000000;
    return { seconds, nanos };
}
function fromTimestamp(t) {
    let millis = (t.seconds || 0) * 1000;
    millis += (t.nanos || 0) / 1000000;
    return new globalThis.Date(millis);
}
function fromJsonTimestamp(o) {
    if (o instanceof globalThis.Date) {
        return o;
    }
    else if (typeof o === "string") {
        return new globalThis.Date(o);
    }
    else {
        return fromTimestamp(Timestamp.fromJSON(o));
    }
}
function isSet(value) {
    return value !== null && value !== undefined;
}
