// @devvit/protos/types/devvit/plugin/externalendpoints/v1alpha/externalendpoints.js
/**
 * #externalendpoints.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
import _m0 from "protobufjs/minimal.js";
import { messageTypeRegistry } from "../../../../typeRegistry.js";
/** External endpoint token / endpoint access permission. */
export var Scope;
(function (Scope) {
    Scope[Scope["SCOPE_UNSPECIFIED"] = 0] = "SCOPE_UNSPECIFIED";
    /**
     * SCOPE_INSTALL - Install allows short-lived callback tokens generated by the app in an
     * outgoing `fetch()` wherever the app is installed (a subreddit or user
     * account).
     */
    Scope[Scope["SCOPE_INSTALL"] = 1] = "SCOPE_INSTALL";
    /** SCOPE_GLOBAL - Global allows managed tokens. */
    Scope[Scope["SCOPE_GLOBAL"] = 2] = "SCOPE_GLOBAL";
    Scope[Scope["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(Scope || (Scope = {}));
export function scopeFromJSON(object) {
    switch (object) {
        case 0:
        case "SCOPE_UNSPECIFIED":
            return Scope.SCOPE_UNSPECIFIED;
        case 1:
        case "SCOPE_INSTALL":
            return Scope.SCOPE_INSTALL;
        case 2:
        case "SCOPE_GLOBAL":
            return Scope.SCOPE_GLOBAL;
        case -1:
        case "UNRECOGNIZED":
        default:
            return Scope.UNRECOGNIZED;
    }
}
export function scopeToJSON(object) {
    switch (object) {
        case Scope.SCOPE_UNSPECIFIED:
            return 0;
        case Scope.SCOPE_INSTALL:
            return 1;
        case Scope.SCOPE_GLOBAL:
            return 2;
        case Scope.UNRECOGNIZED:
        default:
            return -1;
    }
}
function createBaseGetCallbackUrlRequest() {
    return { entry: "" };
}
export const GetCallbackUrlRequest = {
    $type: "devvit.plugin.externalendpoints.v1alpha.GetCallbackUrlRequest",
    encode(message, writer = _m0.Writer.create()) {
        if (message.entry !== "") {
            writer.uint32(10).string(message.entry);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetCallbackUrlRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.entry = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { entry: isSet(object.entry) ? globalThis.String(object.entry) : "" };
    },
    toJSON(message) {
        const obj = {};
        if (message.entry !== "") {
            obj.entry = message.entry;
        }
        return obj;
    },
    create(base) {
        return GetCallbackUrlRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetCallbackUrlRequest();
        message.entry = object.entry ?? "";
        return message;
    },
};
messageTypeRegistry.set(GetCallbackUrlRequest.$type, GetCallbackUrlRequest);
function createBaseGetCallbackUrlResponse() {
    return { callbackUrl: "" };
}
export const GetCallbackUrlResponse = {
    $type: "devvit.plugin.externalendpoints.v1alpha.GetCallbackUrlResponse",
    encode(message, writer = _m0.Writer.create()) {
        if (message.callbackUrl !== "") {
            writer.uint32(10).string(message.callbackUrl);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetCallbackUrlResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.callbackUrl = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { callbackUrl: isSet(object.callbackUrl) ? globalThis.String(object.callbackUrl) : "" };
    },
    toJSON(message) {
        const obj = {};
        if (message.callbackUrl !== "") {
            obj.callbackUrl = message.callbackUrl;
        }
        return obj;
    },
    create(base) {
        return GetCallbackUrlResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetCallbackUrlResponse();
        message.callbackUrl = object.callbackUrl ?? "";
        return message;
    },
};
messageTypeRegistry.set(GetCallbackUrlResponse.$type, GetCallbackUrlResponse);
function createBaseGetExternalUrlRequest() {
    return { appSlug: "", locationId: "" };
}
export const GetExternalUrlRequest = {
    $type: "devvit.plugin.externalendpoints.v1alpha.GetExternalUrlRequest",
    encode(message, writer = _m0.Writer.create()) {
        if (message.appSlug !== "") {
            writer.uint32(10).string(message.appSlug);
        }
        if (message.locationId !== "") {
            writer.uint32(18).string(message.locationId);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetExternalUrlRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.appSlug = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.locationId = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            appSlug: isSet(object.appSlug) ? globalThis.String(object.appSlug) : "",
            locationId: isSet(object.locationId) ? globalThis.String(object.locationId) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.appSlug !== "") {
            obj.appSlug = message.appSlug;
        }
        if (message.locationId !== "") {
            obj.locationId = message.locationId;
        }
        return obj;
    },
    create(base) {
        return GetExternalUrlRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetExternalUrlRequest();
        message.appSlug = object.appSlug ?? "";
        message.locationId = object.locationId ?? "";
        return message;
    },
};
messageTypeRegistry.set(GetExternalUrlRequest.$type, GetExternalUrlRequest);
function createBaseGetExternalUrlResponse() {
    return { externalUrl: "" };
}
export const GetExternalUrlResponse = {
    $type: "devvit.plugin.externalendpoints.v1alpha.GetExternalUrlResponse",
    encode(message, writer = _m0.Writer.create()) {
        if (message.externalUrl !== "") {
            writer.uint32(10).string(message.externalUrl);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetExternalUrlResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.externalUrl = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { externalUrl: isSet(object.externalUrl) ? globalThis.String(object.externalUrl) : "" };
    },
    toJSON(message) {
        const obj = {};
        if (message.externalUrl !== "") {
            obj.externalUrl = message.externalUrl;
        }
        return obj;
    },
    create(base) {
        return GetExternalUrlResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetExternalUrlResponse();
        message.externalUrl = object.externalUrl ?? "";
        return message;
    },
};
messageTypeRegistry.set(GetExternalUrlResponse.$type, GetExternalUrlResponse);
export const ExternalEndpointsServiceName = "devvit.plugin.externalendpoints.v1alpha.ExternalEndpoints";
export class ExternalEndpointsClientImpl {
    constructor(rpc, opts) {
        this.service = opts?.service || ExternalEndpointsServiceName;
        this.rpc = rpc;
        this.GetCallbackUrl = this.GetCallbackUrl.bind(this);
        this.GetExternalUrl = this.GetExternalUrl.bind(this);
    }
    GetCallbackUrl(request, metadata) {
        const data = GetCallbackUrlRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "GetCallbackUrl", data, metadata);
        return promise.then((data) => GetCallbackUrlResponse.decode(_m0.Reader.create(data)));
    }
    GetExternalUrl(request, metadata) {
        const data = GetExternalUrlRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "GetExternalUrl", data, metadata);
        return promise.then((data) => GetExternalUrlResponse.decode(_m0.Reader.create(data)));
    }
}
export const ExternalEndpointsDefinition = {
    name: "ExternalEndpoints",
    fullName: "devvit.plugin.externalendpoints.v1alpha.ExternalEndpoints",
    methods: {
        /**
         * Get a callback URL for the specified endpoint with a short-term token.
         * Eg,
         * `https://wsbapp-2th52-external.devvit.net/external/on/image/generated?externalToken=devvit_at_abc123`.
         * Useful for run-as and any ephemeral usage.
         */
        getCallbackUrl: {
            name: "GetCallbackUrl",
            requestType: GetCallbackUrlRequest,
            requestStream: false,
            responseType: GetCallbackUrlResponse,
            responseStream: false,
            options: {},
        },
        /**
         * Get the root `/external/` URL of an install for managed tokens. Eg,
         * `https://wsbapp-2th52-external.devvit.net/external/`. Useful for
         * install-to-install communication.
         */
        getExternalUrl: {
            name: "GetExternalUrl",
            requestType: GetExternalUrlRequest,
            requestStream: false,
            responseType: GetExternalUrlResponse,
            responseStream: false,
            options: {},
        },
    },
};
function isSet(value) {
    return value !== null && value !== undefined;
}
