// @devvit/protos/json/devvit/platform/v1/request_context.js
/**
 * #request_context.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
export var AppStatus;
(function (AppStatus) {
    /** APP_STATUS_UNSPECIFIED - Default value. Should not be used for requests. */
    AppStatus[AppStatus["APP_STATUS_UNSPECIFIED"] = 0] = "APP_STATUS_UNSPECIFIED";
    /** PRERELEASE - App is in prerelease state */
    AppStatus[AppStatus["PRERELEASE"] = 1] = "PRERELEASE";
    /** PUBLISHED - App is published and available */
    AppStatus[AppStatus["PUBLISHED"] = 2] = "PUBLISHED";
    AppStatus[AppStatus["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(AppStatus || (AppStatus = {}));
/** Provenance of the request */
export var RequestSource;
(function (RequestSource) {
    RequestSource[RequestSource["UNKNOWN"] = 0] = "UNKNOWN";
    /** TRIGGER - Event triggers */
    RequestSource[RequestSource["TRIGGER"] = 1] = "TRIGGER";
    /** SCHEDULER - Scheduled actions, via HandleScheduledAction */
    RequestSource[RequestSource["SCHEDULER"] = 2] = "SCHEDULER";
    /** USER_BLOCKS - User-initiated requests on Blocks-based actions */
    RequestSource[RequestSource["USER_BLOCKS"] = 3] = "USER_BLOCKS";
    /** USER_WEBBIT - User-initiated requests via webbit URLs */
    RequestSource[RequestSource["USER_WEBBIT"] = 4] = "USER_WEBBIT";
    RequestSource[RequestSource["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(RequestSource || (RequestSource = {}));
