// @devvit/protos/json/devvit/plugin/telemetry/telemetry.js
/**
 * #telemetry.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
/** Machine-readable status for a telemetry event attempt. */
export var JourneyReceiptStatus;
(function (JourneyReceiptStatus) {
    /** JOURNEY_RECEIPT_UNSPECIFIED - Status was not provided. */
    JourneyReceiptStatus[JourneyReceiptStatus["JOURNEY_RECEIPT_UNSPECIFIED"] = 0] = "JOURNEY_RECEIPT_UNSPECIFIED";
    /** JOURNEY_RECEIPT_VALID - Event was valid and sent to the telemetry pipeline. */
    JourneyReceiptStatus[JourneyReceiptStatus["JOURNEY_RECEIPT_VALID"] = 1] = "JOURNEY_RECEIPT_VALID";
    /** JOURNEY_RECEIPT_DENIED_NOT_ALLOWLISTED - Event was valid, but not sent because the app is not allowlisted. */
    JourneyReceiptStatus[JourneyReceiptStatus["JOURNEY_RECEIPT_DENIED_NOT_ALLOWLISTED"] = 2] = "JOURNEY_RECEIPT_DENIED_NOT_ALLOWLISTED";
    /** JOURNEY_RECEIPT_DENIED_RATE_LIMITED - Event was valid, but not sent because the app exceeded rate limits. */
    JourneyReceiptStatus[JourneyReceiptStatus["JOURNEY_RECEIPT_DENIED_RATE_LIMITED"] = 3] = "JOURNEY_RECEIPT_DENIED_RATE_LIMITED";
    /** JOURNEY_RECEIPT_DENIED_DUPLICATE - Event was valid, but not sent because it was a duplicate. */
    JourneyReceiptStatus[JourneyReceiptStatus["JOURNEY_RECEIPT_DENIED_DUPLICATE"] = 4] = "JOURNEY_RECEIPT_DENIED_DUPLICATE";
    /** JOURNEY_RECEIPT_INVALID - Event was invalid and not sent to the telemetry pipeline. */
    JourneyReceiptStatus[JourneyReceiptStatus["JOURNEY_RECEIPT_INVALID"] = 5] = "JOURNEY_RECEIPT_INVALID";
    /** JOURNEY_RECEIPT_DENIED_DISABLED - Event was valid, but not sent because Journey telemetry is disabled. */
    JourneyReceiptStatus[JourneyReceiptStatus["JOURNEY_RECEIPT_DENIED_DISABLED"] = 6] = "JOURNEY_RECEIPT_DENIED_DISABLED";
    /** JOURNEY_RECEIPT_DENIED_PLAYTEST - Event was valid, but not sent because the app version is a playtest version. */
    JourneyReceiptStatus[JourneyReceiptStatus["JOURNEY_RECEIPT_DENIED_PLAYTEST"] = 7] = "JOURNEY_RECEIPT_DENIED_PLAYTEST";
    JourneyReceiptStatus[JourneyReceiptStatus["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(JourneyReceiptStatus || (JourneyReceiptStatus = {}));
