// @devvit/public-api/apis/ui/UIClient.js
var __classPrivateFieldGet = (this && this.__classPrivateFieldGet) || function (receiver, state, kind, f) {
    if (kind === "a" && !f) throw new TypeError("Private accessor was defined without a getter");
    if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver)) throw new TypeError("Cannot read private member from an object whose class did not declare it");
    return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
};
var _UIClient_effects;
import { EffectType } from '@devvit/protos/json/devvit/ui/effects/v1alpha/effect.js';
import { ToastAppearance, } from '@devvit/protos/json/devvit/ui/toast/toast.js';
import { resolveNavigationInput } from '@devvit/shared-types/thing-navigation.js';
import { Devvit } from '../../devvit/Devvit.js';
import { assertValidFormFields } from './helpers/assertValidFormFields.js';
import { transformFormFields } from './helpers/transformForm.js';
export class UIClient {
    constructor() {
        _UIClient_effects.set(this, []);
    }
    showForm(formKey, data) {
        return this.showFormInternal(formKey, data);
    }
    showFormInternal(formKey, data, formInternalOverride) {
        const formDefinition = Devvit.formDefinitions?.get(formKey);
        if (!formDefinition) {
            throw new Error('Form does not exist. Make sure you have added it using Devvit.createForm at the root of your app.');
        }
        const formData = formInternalOverride ??
            (formDefinition.form instanceof Function
                ? formDefinition.form(data ?? {})
                : formDefinition.form);
        const form = {
            fields: [],
            id: formKey,
            title: formData.title,
            acceptLabel: formData.acceptLabel,
            cancelLabel: formData.cancelLabel,
            shortDescription: formData.description,
        };
        assertValidFormFields(formData.fields);
        form.fields = transformFormFields(formData.fields);
        __classPrivateFieldGet(this, _UIClient_effects, "f").push({
            type: EffectType.EFFECT_SHOW_FORM,
            showForm: {
                form,
            },
        });
    }
    showToast(textOrToast) {
        let toast;
        if (textOrToast instanceof Object) {
            toast = {
                text: textOrToast.text,
                appearance: textOrToast.appearance === 'success' ? ToastAppearance.SUCCESS : ToastAppearance.NEUTRAL,
            };
        }
        else {
            toast = {
                text: textOrToast,
            };
        }
        __classPrivateFieldGet(this, _UIClient_effects, "f").push({
            type: EffectType.EFFECT_SHOW_TOAST,
            showToast: {
                toast,
            },
        });
    }
    navigateTo(thingOrUrl) {
        const inputUrl = resolveNavigationInput(thingOrUrl);
        if (!URL.canParse(inputUrl)) {
            throw new TypeError(`Invalid URL: ${inputUrl}`);
        }
        const normalizedUrl = new URL(inputUrl).toString();
        __classPrivateFieldGet(this, _UIClient_effects, "f").push({
            type: EffectType.EFFECT_NAVIGATE_TO_URL,
            navigateToUrl: {
                url: normalizedUrl,
            },
        });
    }
    /** @internal */
    get __effects() {
        return __classPrivateFieldGet(this, _UIClient_effects, "f");
    }
}
_UIClient_effects = new WeakMap();
